apps/api/src/expenses/expenses.controller.ts

```diff
@@ -46,11 +62,60 @@ export class ExpensesController {
   }
 
   @Delete(':id')
+  remove(@Param('id') id: string) {
+    return this.expensesService.remove(id)
+  }
+
+  @Get('campaign/:slug')
   @Roles({
     roles: [RealmViewSupporters.role, ViewSupporters.role],
     mode: RoleMatchingMode.ANY,
   })
-  remove(@Param('id') id: string) {
-    return this.expensesService.remove(id)
+  async listCampaignExpenses(@Param('slug') slug: string) {
+    return this.expensesService.listCampaignExpenses(slug)
+  }
+
+  @Get('campaign/approved/:slug')
+  @Public()
+  async listCampaignApprovedExpenses(@Param('slug') slug: string) {
+    return this.expensesService.listCampaignApprovedExpenses(slug)
+  }
+
+  @Post('upload-files/:id')
```

Comment:
the upload-files/:id looks strange and it is not clear if this is a fileID or expenseID.
If expenseId then the route should be expenses/:id/files and the POST will instruct it is uploading

Reply:
ok. fixed it.